nominal/core/dataset.py

```diff
@@ -373,6 +373,14 @@ def set_channel_prefix_tree(self, delimiter: str = ".") -> None:
         request = datasource_api.IndexChannelPrefixTreeRequest(self.rid, delimiter=delimiter)
         self._clients.datasource.index_channel_prefix_tree(self._clients.auth_header, request)
 
+    def archive(self) -> None:
+        """Archives the dataset, preventing it from showing up in the 'All Datasets' pane in the UI."""
```

Comment:
nit: would prefer to leave most of the "this has X behavior in the app" out since that's subject to change without any relation to the client lib